README.md

```diff
@@ -20,24 +20,28 @@ This project has the following goals:
 
 ## Roadmap
 
-Below is a rough summary of what features work and don't work.
-Some of those features are a likely outside the scope of this project, we are still learning and exploring what's possible.
-Please share your thoughts in
-[#2](https://github.com/scalameta/language-server/issues/2).
-
-- [x] Linting with Scalafix
-- [x] Formatting with Scalafmt
-- [x] Auto completions as you type with presentation compiler
-- [x] Go to definition from project Scala sources to project Scala sources
-- [x] Show type at position
-- [x] Go to definition from project sources to Scala dependency source files
-- [x] Go to definition from project sources to Java dependency source files
-- [ ] Show red squigglies as you type
-- [ ] Show red squigglies on compile
-- [ ] Show parameter list as you type, signature helper
-- [ ] Find symbol references
-- [ ] Show docstring on hover
-- [ ] Rename symbol
+Below is a rough summary of what features have been implemented.
+Even if some checkbox is marked it does not mean that feature works perfectly.
+Some of those features are a likely outside the scope of this project, we are
+still learning and exploring what's possible.
+
+- [x] Linting with Scalafix on compile (textDocument/publishDiagnostics)
+- [ ] Linting with Scalafix as you type (textDocument/publishDiagnostics)
+- [x] Formatting with Scalafmt (textDocument/formatting)
```

Comment:
while we're at it, we should also add `textDocument/rangeFormatting` and `textDocument/onTypeFormatting`

Reply:
range formatting is beyond the scope of this project, I spent a long time trying to support it in scalafmt and didn't get it working

Reply:
I thought it was working. I use it in vscode-scalafmt and it seems to be ok if you try to format a "reasonable" range of statements.

Reply:
Cool, I added it. Format on type looks interesting, I wasn't aware that was a thing!